### PR TITLE
Add @serializabe_property

### PR DESCRIPTION
**Pull request by mrijk**

### CODE REVIEW

**Review by tjeerddie — CHANGES_REQUESTED** [review]
Add tests

File: orchestrator/domain/base.py [hunk 1]
```diff
@@ -63,6 +63,12 @@ class ProductNotInRegistryException(Exception):
     pass
 
 
+class serializabe_property(property):
```
Comment (dmarjenburgh) [inline orchestrator/domain/base.py:66]:
```suggestion
class serializable_property(property):
```

File: orchestrator/domain/base.py [hunk 2]
```diff
@@ -416,6 +422,47 @@ def _save_instances(
...
+    def dict(
+        self,
+        *,
+        include: Any | None = None,
+        exclude: Any | None = None,
+        by_alias: bool = False,
+        skip_defaults: bool | None = None,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
```
Comment (Mark90) [inline orchestrator/domain/base.py:441]: Is it possible to use kwargs instead?

```python
def dict(self, *, **kwargs) -> dict[str, Any]:
    attribs = super().dict(**kwargs)
    
    if include := kwargs.get('include'):
        ...
```

Comment (mrijk) [reply orchestrator/domain/base.py:441]: Probably. Was just following the suggestion from one of the comments in a Pydantic issue. I'm fine with both options.

**Review by tjeerddie — APPROVED** [review]
LGTM

File: orchestrator/domain/base.py (hunk 2 above)
Comment (mrijk) [reply orchestrator/domain/base.py]: Changed

**Pull request merged**